addons/product/models/product_description_mixin.py

```diff
@@ -0,0 +1,90 @@
+
+
+class ProductDescriptionMixin(models.AbstractModel):
+    _name = "product.description.mixin"
+    _description = 'Product description Mixin'
+
+    product_id = fields.Many2one(
+        'product.product', 'Product')
+    product_custom_attribute_value_ids = fields.Many2many(
+        comodel_name='product.attribute.custom.value',
+        string="Custom Values",
+        compute='_compute_custom_attribute_values',
+        store=True, readonly=False, precompute=True, copy=True)
+    # M2M holding the values of product.attribute with create_variant field set to 'no_variant'
+    # It allows keeping track of the extra_price associated to those attribute values and add them to the SO line description
+    product_no_variant_attribute_value_ids = fields.Many2many(
+        comodel_name='product.template.attribute.value',
+        string="Extra Values",
+        compute='_compute_no_variant_attribute_values',
+        store=True, readonly=False, precompute=True, ondelete='restrict')
+
+    @api.depends('product_id')
+    def _compute_custom_attribute_values(self):
+        for line in self:
+            if not line.product_id:
+                line.product_custom_attribute_value_ids = False
+                continue
+            if not line.product_custom_attribute_value_ids:
+                continue
+            valid_values = line.product_id.product_tmpl_id.valid_product_template_attribute_line_ids.product_template_value_ids
+            # remove the is_custom values that don't belong to this template
+            for pacv in line.product_custom_attribute_value_ids:
+                if pacv.custom_product_template_attribute_value_id not in valid_values:
+                    line.product_custom_attribute_value_ids -= pacv
+
+    @api.depends('product_id')
+    def _compute_no_variant_attribute_values(self):
+        for line in self:
+            if not line.product_id:
+                line.product_no_variant_attribute_value_ids = False
+                continue
+            if not line.product_no_variant_attribute_value_ids:
+                continue
+            valid_values = line.product_id.product_tmpl_id.valid_product_template_attribute_line_ids.product_template_value_ids
+            # remove the no_variant attributes that don't belong to this template
+            for ptav in line.product_no_variant_attribute_value_ids:
+                if ptav._origin not in valid_values:
+                    line.product_no_variant_attribute_value_ids -= ptav
+
+    def _get_multiline_description(self):
+        """ Compute a default multiline description.
+        In most cases the product description is enough.
+        e.g:
+        - custom attributes and attributes that don't create variants, both introduced by the "product configurator"
+        """
+        self.ensure_one()
+        return self.product_id.display_name + self.get_multiline_description_variants()
+
+    def get_multiline_description_variants(self, space=True):
```

Comment:
```suggestion
    def _get_multiline_description_variants(self, space=True):
```
Please keep it private unless necessary

Reply:
The commit and pr description are not updated yet as this pr is not ready yet (I changed it's state to get a first global review of your team to check if I'm going in the right direction), that's why you had no way to see that I'm using it in a orm call in the enterprise pr. 
Anyway, my bad, I'll update all the descriptions and stuff rn